src/blk/BlockDevice.h

```diff
@@ -183,6 +183,7 @@ class BlockDevice {
   bool support_discard = false;
   bool rotational = true;
   bool lock_exclusive = true;
+  bool need_notify = false;
```

Comment:
May be a bit paranoic but I'd make it atomic.

Reply:
In the latest update, the modifications to 'need_notify' are all under the protection of 'discard_lock'. Would it be more graceful to cancel atomic?